config/main.py

```diff
@@ -33,6 +35,70 @@ def run_command(command, display_cmd=False, ignore_error=False):
     if proc.returncode != 0 and not ignore_error:
         sys.exit(proc.returncode)
 
+def interface_alias(interface_name):
+    """Return default interface name if alias name is given as argument
+    """
+
+    cmd = 'sonic-cfggen -d --var-json "PORT"'
+    p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+
+    port_dict = json.loads(p.stdout.read())
+
+    if interface_name is not None:
+        for port_name in natsorted(port_dict.keys()):
+            if interface_name == port_dict[port_name]['alias']:
+                return port_name
+        print "Invalid interface {}".format(interface_name)
+        sys.exit(0)
+
+def interface_name_to_alias(interface_name):
+    """Return alias interface name if default name is given as argument
+    """
+
+    cmd = 'sonic-cfggen -d --var-json "PORT"'
+    p = subprocess.Popen(cmd, shell=True, stdout=subprocess.PIPE)
+
+    port_dict = json.loads(p.stdout.read())
+
+    if interface_name is not None:
+        for port_name in natsorted(port_dict.keys()):
+            if interface_name == port_name:
+                return port_dict[port_name]['alias']
+        print "Invalid interface {}".format(interface_name)
+        sys.exit(0)
```

Comment:
```
def shutdown(interface_name, verbose):
    """Shut down interface"""
    print "lias"
    if get_interface_mode() == "alias":
        interface_name = interface_alias(interface_name)
```
- If we fail to return sys.exit(0) in function interface_alias() it propogates to next line and execute 
"ip link set <interface_nam> down. Where interface_name is invalid.
- This is unnecessary and tends to double opertaion. To avoid this sys.exit(0) introduced. and also it alerts the user about invalid interface.
- If you are skeptical to the exit in  utiltiy function we can shift the exit in parent shutdown().

```
    command = "ip link set {} down".format(interface_name)
    run_command(command, display_cmd=verbose)

```

#

Reply:
Yes. I think the caller functions should check for `None`, print an error and exit.

Reply:
Will addres the changes